prompting/tasks/base_task.py

```diff
@@ -24,54 +28,93 @@ def CHATTENSOR_SYSTEM_PROMPT():
 
+    @model_validator(mode="after")
+    def get_model_id_and_seed(self) -> "BaseTextTask":
+        if self.llm_model:
+            self.llm_model_id = self.llm_model.llm_model_id if self.llm_model else None
+            self.seed = random.randint(0, 1000000)
+        return self
+
+    @abstractmethod
+    def make_query(self, dataset_entry: DatasetEntry, **kwargs) -> str:
+        raise NotImplementedError("Method generate_query_reference must be implemented")
+
     @abstractmethod
-    def generate_query_reference(llm_pipeline: BasePipeline, context: Context, **kwargs) -> [str, str]:
+    def make_reference(self, dataset_entry: DatasetEntry) -> str:
         raise NotImplementedError("Method generate_query_reference must be implemented")
 
-    @classmethod
-    def generate_reference(cls, llm_pipeline: BasePipeline, messages: list[str]) -> str:
+    def generate_query_reference(self, dataset_entry: DatasetEntry) -> str:
+        self.make_query(dataset_entry=dataset_entry)
+        self.make_reference(dataset_entry=dataset_entry)
+        return self.query, self.reference
+
+    def generate_reference(self, messages: list[str]) -> str:
         """Generates a reference answer to be used for scoring miner completions"""
         logger.info("🤖 Generating reference...")
-        reference = vLLM_LLM(llm_pipeline, system_prompt=cls.reference_system_prompt or "").query(
-            cleaner=cls.cleaner, message=messages
-        )
-        return reference
+        self.reference = vLLM_LLM(
+            llm=model_manager.get_model(self.llm_model), system_prompt=self.reference_system_prompt or ""
+        ).query(cleaner=self.cleaner, message=messages)
+        if self.reference is None:
+            raise Exception("Reference generation failed")
+        return self.reference
 
-    @classmethod
     def generate_query(
-        cls,
-        message: str,
-        llm_pipeline: BasePipeline,
+        self,
+        messages: str,
     ) -> str:
         """Generates a query to be used for generating the challenge"""
         logger.info("🤖 Generating query...")
-        query = vLLM_LLM(llm_pipeline, system_prompt=cls.query_system_prompt or "").query(message=message)
-        return cls.augment_query(query, llm_pipeline)
+        self.query = vLLM_LLM(
+            llm=model_manager.get_model(self.llm_model), system_prompt=self.query_system_prompt or ""
+        ).query(message=messages)
+        if self.query is None:
+            raise Exception("Query generation failed")
+        return self.augment_query(self.query)
 
-    @classmethod
     def augment_query(
-        cls,
+        self,
         query: str,
-        llm_pipeline: BasePipeline,
     ) -> str:
         """Creates the opening question of the conversation which is based on the task query but dressed in the persona of the user."""
-        if not cls.augmentation_system_prompt:
+        if self.augmentation_system_prompt:
```

Comment:
Important, it will break all the challenges and references
```suggestion
        if not self.augmentation_system_prompt:
```

Reply:
Again good spot, massive thanks